### PR TITLE
Process +build flags (fixes #1199)

### LINKED ISSUE #1199: [1.0] support +build flag
```go
package main

import "net"

var _ net.TCPListener

func main() {
}
```
Unresolved type 'TCPListener'

plugin: 1.0.0.alpha#11
os: windows
idea 14.0.3 (IC-139.1117)

### PR DESCRIPTION
**Pull request by linniksa**
fixes #1199

Also currently two files in same directory (gen1, get2 for example)

```go
// +build ignore

package main;

func test() {

}
```
produce test function is duplicates

### CODE REVIEW

**Comment by zolotov** [comment]
Looks nice, thanks. I think we should extract matching to separate class that will accept target os and build-constraints. So it will be easy to test isolated class and GoUtil won't be so big :)

**Comment by zolotov** [comment]
Also please keep the logic with go.target.os property until there is no UI for that.

File: src/com/goide/util/GoUtil.java [hunk 1]
```diff
@@ -49,38 +49,180 @@
...
+    // TODO support .c, .cpp and other
+    if (!(file instanceof GoFile)) {
+      return false;
+    }
+
+    // Look for +build comments to accept or reject the file.
+    // TODO support for custom build tags
+    String flags = ((GoFile)file).getBuildFlags();
+
+    if (null == flags || flags.isEmpty()) {
```
Comment (zolotov) [inline src/com/goide/util/GoUtil.java]: Cannot be empty but anyway: StringUtil.isEmpty checks both conditions.

**Comment by linniksa** [comment]
done

**Pull request merged**